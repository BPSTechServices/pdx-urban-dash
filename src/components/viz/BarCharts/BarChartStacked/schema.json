{
  "$schema": "https://vega.github.io/schema/vega/v5.json",
  "width": 200,
  "height": 200,
  "padding": 5,

  "data": [
    {
      "name": "table",
      "$dataValues": "",
      "transform": [
        {
          "type": "stack",
          "groupby": ["x"],
          "sort": {"data": "table", "field": "c"},
          "field": "y"
        }
      ]
    }
  ],

  "config": {
    "axis": {
      "$labelColor": "",
      "$labelFontSize": 0
    }
  },

  "scales": [
    {
      "name": "x",
      "type": "band",
      "domain": {"data": "table", "field": "x"},
      "range": "width",
      "padding": 0.05,
      "round": true
    },
    {
      "name": "y",
      "domain": {"data": "table", "field": "y1"},
      "range": "height",
      "nice": true,
      "zero": true
    },
    {
      "name": "color",
      "type": "ordinal",
      "$colorRange": [],
      "domain": {"data": "table", "field": "c"}
    }
  ],

  "axes": [
    { "orient": "bottom", "scale": "x" },
    { "orient": "left", "scale": "y" }
  ],

  "marks": [
    {
      "type": "group",
      "from": {
        "facet": {
          "name": "facet",
          "data": "table",
          "groupby": "c"
        }
      },

      "marks": [
        {
          "name": "bars",
          "from": {"data": "facet"},
          "type": "rect",
          "encode": {
            "enter": {
              "x": {"scale": "x", "field": "x"},
              "y": {"scale": "y", "field": "y0"},
              "y2": {"scale": "y", "field": "y1"},

              "width": {"scale": "x", "band": 1},
              "fill": {"scale": "color", "field": "c"}
            },
            "update": {
              "fillOpacity": {"value": 1}
            },
            "hover": {
              "fillOpacity": {"value": 0.5}
            }
          }
        }
      ]
    }
  ]
}