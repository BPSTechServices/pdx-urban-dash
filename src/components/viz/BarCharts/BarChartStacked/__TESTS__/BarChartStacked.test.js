import React from 'react';
import { shallow } from 'enzyme';
import BarChartStacked from '../index';

const testData = [
  {
    dataSetName: 'first',
    description: 'this is first',
    values: [
      { x: 0, y: 10 },
      { x: 1, y: 20 },
      { x: 2, y: 15 },
      { x: 3, y: 40 },
    ],
  },
  {
    dataSetName: 'second',
    description: 'this is second',
    values: [
      { x: 0, y: 20 },
      { x: 1, y: 30 },
      { x: 2, y: 19 },
      { x: 3, y: 50 },
    ],
  },
];

const testTitle = 'chart title';
const testIcon = 'trending-up';
const testIconSize = 'md';

describe('<BarChartStacked />', () => {
  test('snapshot', () => {
    const wrapper = shallow(<BarChartStacked data={testData} title={testTitle} icon_type={testIcon} icon_size={testIconSize} />);
    expect(wrapper).toMatchSnapshot();
  });
});
