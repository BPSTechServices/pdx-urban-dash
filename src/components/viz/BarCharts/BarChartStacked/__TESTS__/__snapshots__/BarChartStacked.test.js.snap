// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarChartStacked /> snapshot 1`] = `ShallowWrapper {}`;
